habitat_baselines/common/utils.py

```diff
@@ -174,3 +222,111 @@ def generate_video(
         tb_writer.add_video_from_np_images(
             f"episode{episode_id}", checkpoint_idx, images, fps=fps
         )
+
+
+def image_resize_shortest_edge(img, size: int, channels_first: bool = False):
+    """Resizes an img so that the shortest side is length of size.
+
+    Args:
+        img: the array object that needs to be resized (HWC) or (NHWC)
+        size: the size that you want the shortest edge to be resize to
+        channels_first: a boolean that specifies the img is (CHW) or (NCHW)
+    Returns:
+        The resized array as a torch tensor.
+    """
+    no_batch_dim = len(img.shape) == 3
+    if len(img.shape) < 3 or len(img.shape) > 5:
+        raise NotImplementedError()
+    img = _to_tensor(img)
+    if no_batch_dim:
+        img = img.unsqueeze(0)  # Adds a batch dimension
+    if channels_first:
+        # NCHW
+        h, w = img.shape[-2:]
+    else:
+        # NHWC
+        h, w = img.shape[-3:-1]
+        if len(img.shape) == 4:
+            img = img.permute(0, 3, 1, 2)
+        else:
+            img = img.permute(0, 1, 4, 2, 3)
+
+    if w > h:
+        percent = size / h
```

Comment:
what is `percent` here?

Reply:
its the percentage resize to maintain the proper aspect ratio.